contracts/cw4-group/src/helpers.rs

```diff
@@ -0,0 +1,62 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+use cosmwasm_std::{to_binary, Api, CanonicalAddr, CosmosMsg, HumanAddr, StdResult, WasmMsg};
+use cw4::{Cw4Contract, Member};
+
+use crate::msg::HandleMsg;
+
+/// Cw4GroupContract is a wrapper around HumanAddr that provides a lot of helpers
+/// for working with cw4-group contracts.
+///
+/// It extends Cw4Contract to add the extra calls from cw4-group.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Cw4GroupContract(pub Cw4Contract);
+
+impl Deref for Cw4GroupContract {
+    type Target = Cw4Contract;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl Cw4GroupContract {
+    pub fn new(addr: HumanAddr) -> Self {
+        Cw4GroupContract(Cw4Contract(addr))
+    }
+
+    /// Convert this address to a form fit for storage
+    pub fn canonical(&self, api: &dyn Api) -> StdResult<Cw4GroupCanonicalContract> {
+        let canon = api.canonical_address(&self.addr())?;
+        Ok(Cw4GroupCanonicalContract(canon))
+    }
```

Comment:
I don't understand why the wrappers are needed. Why not return a `CanonicalAddr` directly?

Why not a `to_canonical()` method in `HumanAddr`?

Reply:
> Why not a to_canonical() method in HumanAddr?

Because it requires the `deps.api` to do the transformation. We could add it but it would not be any simpler.

 > I don't understand why the wrappers are needed. Why not return a CanonicalAddr directly?

I was storing everything as `CanonicalAddr` in state. I see even the sdk is moving to persist as bech32 strings, and maybe we can start removing this canonical transform at some point.

Reply:
> > Why not a to_canonical() method in HumanAddr?
> 
> Because it requires the `deps.api` to do the transformation. We could add it but it would not be any simpler.

Yeah, you're right, passing the api to the method doesn't look any better.

> 
> > I don't understand why the wrappers are needed. Why not return a CanonicalAddr directly?
> 
> I was storing everything as `CanonicalAddr` in state. I see even the sdk is moving to persist as bech32 strings, and maybe we can start removing this canonical transform at some point.

That would be a good thing:tm:.